packages/main/src/plugin/layout-registry.ts

```diff
@@ -0,0 +1,176 @@
+    try {
+      // Auto-register the layout if not already registered
+      if (!REGISTERED_LAYOUTS[layoutKind]) {
+        REGISTERED_LAYOUTS[layoutKind] = availableColumns;
+
+        // Register the configuration with the configuration registry
+        const layoutConfiguration: IConfigurationNode = {
+          id: 'preferences',
+          title: 'Layout',
+          type: 'object',
+          properties: {
+            [`layout.${layoutKind}`]: {
+              description: `Preferred layout of columns for ${layoutKind}`,
+              type: 'array',
+              default: this.parseConfiguration(availableColumns),
+              hidden: true,
+            },
+          },
+        };
+
+        this.#disposables.push(this.configurationRegistry.registerConfigurations([layoutConfiguration]));
+      }
+
+      const config = this.configurationRegistry.getConfiguration('layout');
+      const savedConfig = config.get<SavedLayoutConfig[]>(`${layoutKind}`, []);
+
+      if (!savedConfig || !Array.isArray(savedConfig)) {
+        return this.createDefaultLayoutItems(layoutKind, availableColumns);
+      }
+
+      return this.mergeConfigWithAvailableColumns(layoutKind, savedConfig, availableColumns);
+    } catch (error: unknown) {
+      console.warn(`Failed to load layout config for ${layoutKind}:`, error);
+      return this.createDefaultLayoutItems(layoutKind, availableColumns);
+    }
+  }
+
+  // Save layout configuration
+  async saveLayoutConfig(layoutKind: string, items: LayoutEditItem[]): Promise<void> {
+    const configToSave: SavedLayoutConfig[] = items.map(item => ({ id: item.id, enabled: item.enabled }));
+
+    const config = this.configurationRegistry.getConfiguration('layout');
+    await config.update(`${layoutKind}`, configToSave);
+  }
+
+  // Reset layout to defaults
+  async resetLayoutConfig(layoutKind: string, availableColumns: string[]): Promise<LayoutEditItem[]> {
+    const config = this.configurationRegistry.getConfiguration('layout');
+    await config.update(`${layoutKind}`, undefined);
+    return this.createDefaultLayoutItems(layoutKind, availableColumns);
+  }
+
+  // Helper: Create default layout items
+  private createDefaultLayoutItems(layoutKind: string, availableColumns: string[]): LayoutEditItem[] {
+    const defaults = REGISTERED_LAYOUTS[layoutKind] ?? availableColumns;
+    return defaults.map((colName, index) => ({
+      id: colName,
+      label: this.getColumnLabel(layoutKind, colName),
+      enabled: true,
+      originalOrder: index,
+    }));
+  }
+
+  // Helper: Get proper label for column (dashboard sections have different labels)
+  private getColumnLabel(tableKind: string, columnId: string): string {
+    if (tableKind === 'dashboard') {
+      // Transform kebab-case to Title Case (e.g., 'release-notes' -> 'Release Notes')
+      return columnId
+        .split('-')
+        .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+        .join(' ');
+    }
+    return columnId;
```

Comment:
it can't contain something specific like 'dashboard'

it needs to be generic. Like format of the column to use.

Then we choose the strategy 'kebab-case', 'camelCase', etc from frontend but it's not the backend that knows 'dashboard'